### PR TITLE
improve research by artist

### PR DESCRIPTION
**Pull request by vincentDcmps**
I have added specific function for looking for an artist without that it seems search seem to look in all field and not only in artist field

### CODE REVIEW

File: mopidy_subidy/subsonic_api.py [hunk 1]
```diff
@@ -410,6 +410,25 @@ def get_raw_songs(self, album_id):
             return songs
         return []
 
+    def get_raw_random_song(self, size=MAX_LIST_RESULTS):
+        try:
+            response = self.connection.getRandomSongs(size)
+        except Exception:
+            logger.warning(
+                "Connecting to subsonic failed when loading list of songs in album."
```
Comment (Prior99) [inline mopidy_subidy/subsonic_api.py:418]: This error message is misleading.

File: mopidy_subidy/library.py [hunk 2]
```diff
@@ -138,15 +145,34 @@ def search_by_artist_album_and_track(
...
-        artists = self.subsonic_api.get_raw_artists()
-        artist = next(
-            item for item in artists if artist_name in item.get("name")
-        )
-        albums = self.subsonic_api.get_raw_albums(artist.get("id"))
-        album = next(item for item in albums if album_name in item.get("title"))
-        return SearchResult(
-            tracks=self.subsonic_api.get_songs_as_tracks(album.get("id"))
-        )
+        artists = self.subsonic_api.find_raw(artist_name).get("artist")
```
Comment (Prior99) [inline mopidy_subidy/library.py:148]: I can't quite grasp this change. Why was this changed?

File: mopidy_subidy/subsonic_api.py (hunk 1 above)
Comment (vincentDcmps) [reply mopidy_subidy/subsonic_api.py:418]: i will correct this

File: mopidy_subidy/library.py (hunk 2 above)
Comment (vincentDcmps) [reply mopidy_subidy/library.py:148]: Purpose if this change is ti search only album if the artiste search

**Comment by Prior99** [comment]
Hey, thanks again for your patience. Once you resolve the merge conflicts, I will merge and publish this.

**Comment by vincentDcmps** [comment]
done

**Pull request merged**